ocaml/xapi/repository_helpers.ml

```diff
@@ -104,13 +104,13 @@ module Pkg = struct
      *   "qemu-dp-2:2.12.0-2.0.11.x86_64", or
      *   "time-(none):1.7-45.el7.x86_64".
      * These may come from "yum updateinfo list updates" and "rpm -qa" *)
-    let r = Re.Str.regexp "^\\(.+\\)\\.\\(noarch\\|x86_64\\)$" in
-    match Re.Str.string_match r s 0 with
-    | true -> (
+    let r = Re.Posix.compile_pat {|^(.+).(noarch|x86_64)$|} in
```

Comment:
In the original, the dot before `noarch` is escaped - it is not here - so it is not a literal dot.

Reply:
This is unit-tested, I'm suprised it doesn't provoke any failures

Reply:
Added a test that detects the failure